### PR TITLE
HHH-17687 AttributeConverter, query does not use converter to convert 'null' fields

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-17687

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java [hunk 1]
```diff
@@ -377,33 +362,48 @@ else if ( domainParamBinding.getBindType() instanceof BasicValuedMapping ) {
 					final BasicValueConverter valueConverter = jdbcMapping == null ? null : jdbcMapping.getValueConverter();
 					if ( valueConverter != null ) {
 						final Object convertedValue = valueConverter.toRelationalValue( domainParamBinding.getBindValue() );
+						if ( convertedValue != null ) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java:365]: Maybe I'm missing something, but wouldn't this become a problem if e.g. `"val1"` is converted to `null` i.e. `domainParamBinding.getBindValue()` is `"val1"` and `convertedValue` is `null`, leading to the creation of a `JdbcParameterBindingImpl` with the unconverted value?

File: hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java [hunk 2]
```diff
@@ -377,33 +362,48 @@ else if ( domainParamBinding.getBindType() instanceof BasicValuedMapping ) {
...
-							final JdbcParameter jdbcParameter = jdbcParams.get( 0 );
-							jdbcParameterBindings.addBinding(
-									jdbcParameter,
-									new JdbcParameterBindingImpl( jdbcMapping, convertedValue )
-							);
+							for ( int j = 0; j < jdbcParams.size(); j++ ) {
+								final JdbcParameter jdbcParameter = jdbcParams.get( j );
+								jdbcParameterBindings.addBinding(
+										jdbcParameter,
+										new JdbcParameterBindingImpl( null, null )
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java:387]: I'm actually also curious if it weren't better to pass a non-null `jdbcMapping`, even if the `bindValue` is `null`.

```suggestion
										new JdbcParameterBindingImpl( jdbcMapping, null )
```

File: hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java (hunk 1 above)
Comment (beikov) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java:365]: Now we'd miss a `JdbcParameterBindingImpl` for values that get converted to `null`.

Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java:365]: :( going to fix this.

**Review by beikov — APPROVED** [review]

**Pull request merged**